src/vstest.console/CommandLine/InferHelper.cs

```diff
@@ -26,84 +26,128 @@ internal InferHelper(IAssemblyMetadataProvider assemblyMetadataProvider)
     /// <summary>
     /// Determines Architecture from sources.
     /// </summary>
-    public Architecture AutoDetectArchitecture(IList<string> sources, IDictionary<string, Architecture> sourcePlatforms, Architecture defaultArchitecture)
+    public Architecture AutoDetectArchitecture(IList<string> sources, Architecture defaultArchitecture, out IDictionary<string, Architecture> sourceToPlatformMap)
     {
-        var architecture = defaultArchitecture;
+        sourceToPlatformMap = new Dictionary<string, Architecture>();
+        if (sources == null || sources.Count == 0)
+            return defaultArchitecture;
+
+        // Set the default for all sources.
+        foreach (var source in sources)
+        {
+            // TODO: Add default architecture to runtime providers info, or something and that will allow us to have test
+            // cases without any sources. Otherwise change test AutoDetectArchitectureShouldReturnDefaultArchitectureOnNullItemInSources
+            // because this condition is making that test happy.
+            if (source != null)
+            {
+                sourceToPlatformMap.Add(source, defaultArchitecture);
```

Comment:
Are we sure we don't have duplicate sources?

Reply:
Yes, pretty sure.